### PR TITLE
Add integration test for creation of a device and all prerequisite models

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes #DNE
# What's Changed

This is an integration test to hopefully prevent issues like #6441 from being introduced again.

This also fixes a bug in the existing navbar integration tests where we're doing an incorrect `assertTrue(bool("true"))` which would also pass for the negative case of `assertTrue(bool("false"))`

# Screenshots

https://github.com/user-attachments/assets/8a283cab-eaf8-4e35-bd4b-2882f927d4fb

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/tests/integration/test_create_device.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+from nautobot.core.testing.integration import SeleniumTestCase
+
+
+class CreateDeviceTestCase(SeleniumTestCase):
+    """
+    Create a device and all pre-requisite objects through the UI.
+    """
+
+    def _click_nav_bar(self, parent_menu_name, child_menu_name):
```
Comment (glennmatthews) [inline nautobot/dcim/tests/integration/test_create_device.py:12]: Might make sense for this (and maybe some of the other helpers below) to be added to `SeleniumTestCase` so that we can later refactor some of the other integration tests to use them?

**Review by glennmatthews — COMMENTED** [review]
Thanks for adding this!

File: nautobot/dcim/tests/integration/test_create_device.py (hunk 1 above)
Comment (gsnider2195) [reply nautobot/dcim/tests/integration/test_create_device.py:12]: That sounds good to me. I'm happy with how these helper functions turned out

File: nautobot/dcim/tests/integration/test_create_device.py [hunk 2]
```diff
@@ -87,52 +24,61 @@ def test_create_device(self):
...
-        self._click_add_button()
+        self.click_list_view_add_button()
         self.assertEqual(self.browser.url, self.live_server_url + reverse("dcim:device_add"))
         self.browser.fill("name", "Test Device Integration Test 1")
-        self._fill_select2_field("role", "")  # pick first role
-        self._fill_select2_field("device_type", "Test Device Type 1")
-        self._fill_select2_field("location", "Test Location 1")
-        self._fill_select2_field("status", "")  # pick first status
-        self._click_create_button()
+        self.fill_select2_field("role", "")  # pick first role
```
Comment (glennmatthews) [inline nautobot/dcim/tests/integration/test_create_device.py:77]: Might as well use the `Test Role 1`?

**Review by glennmatthews — APPROVED** [review]
